crates/sui-graphql-rpc/src/context_data/db_data_provider.rs

```diff
@@ -1051,6 +1057,134 @@ impl PgManager {
             .await?;
         SuiSystemStateSummary::try_from(result)
     }
+
+    pub(crate) async fn fetch_staked_sui(
+        &self,
+        address: SuiAddress,
+        first: Option<u64>,
+        after: Option<String>,
+        last: Option<u64>,
+        before: Option<String>,
+    ) -> Result<Option<Connection<String, Stake>>, Error> {
+        let obj_filter = ObjectFilter {
+            package: None,
+            module: None,
+            ty: Some(MoveObjectType::staked_sui().to_string()),
+            owner: Some(address),
+            object_ids: None,
+            object_keys: None,
+        };
+        let system_state = self.fetch_latest_sui_system_state().await?;
+        let current_epoch_id = system_state.epoch_id;
+        // we want the object ids of staked sui
+        // The reason we're using this API and not get_staked_by_owner in the
+        // governance api is because that API is
+        // is limited to 1000 objects, and ideally, we want to paginate these stakes
+        let objs = self
+            .multi_get_objs(first, after, last, before, Some(obj_filter))
+            .await?;
+
+        if let Some((stored_objs, has_next_page)) = objs {
+            let mut connection = Connection::new(false, has_next_page);
+
+            let mut edges = vec![];
+
+            for stored_obj in stored_objs {
+                let object = sui_types::object::Object::try_from(stored_obj).map_err(|_| {
+                    Error::Internal("Error converting from stored obj to object".to_string())
+                })?;
+                let stake_object = StakedSui::try_from(&object).map_err(|_| {
```

Comment:
Once we've converted into StakedSui, we can directly leverage get_delegated_stakes

Reply:
Yes, though that is the one function that as of now has some perf issues. Let's revisit this later after I chat with @bmwill.